src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -69,19 +80,52 @@ internal override object GetValueAsObject(object obj, JsonSerializerOptions opti
                 return obj;
             }
 
-            Debug.Assert(Get != null);
+            Debug.Assert(HasGetter);
             return Get((TClass)obj);
         }
 
         internal override void SetValueAsObject(object obj, object value, JsonSerializerOptions options)
         {
-            Debug.Assert(Set != null);
+            Debug.Assert(HasSetter);
             TDeclaredProperty typedValue = (TDeclaredProperty)value;
 
             if (typedValue != null || !IgnoreNullPropertyValueOnWrite(options))
             {
-                Set((TClass)obj, (TDeclaredProperty)value);
+                Set(obj, typedValue);
             }
         }
+
+        internal TDeclaredProperty Get(object obj)
```

Comment:
Can you make this and `Set` private - I wouldn't want these accidently called since they are the boxed \ loosely-typed versions.